src/preact/base-element.js

```diff
@@ -780,13 +787,16 @@ function collectProps(Ctor, element, ref, defaultProps, mediaQueryProps) {
       }
     }
     if (value == null) {
-      if (def.default !== undefined) {
-        props[name] = def.default;
+      const defValue = def.default ?? (def.type && TYPE_DEFAULTS[def.type]);
```

Comment:
I don't understand why we're providing a default "default value".

Reply:
This is not strictly necessary. But so far it has been pretty convenient to pass explicit `true` and `false` values to the React vs going with defaults. I added it here for compatibility with the old implementation, but I can remove this, if it looks too distracting. As the code reads, however, there's a property default and a type default.

Reply:
My issue is that it'd be really common for Preact to just omit the field. Eg, if there were a `disabled`, then receiving all `<C disabled={false}>`, `<C disabled={true}>`, and `<C>` would be normal. If we normalize in the AMP layer, then we'd not receive the `undefined` case.

Reply:
Sounds good. I'll reconcile them.

Reply:
Done. Thanks for the advice.